crates/core/tedge_mapper/src/collectd_mapper/collectd.rs

```diff
@@ -188,20 +213,37 @@ mod tests {
 
         let collectd_message = CollectdMessage::parse_from(&mqtt_message).unwrap();
 
-        let CollectdMessage {
-            metric_group_key,
-            metric_key,
-            timestamp,
-            metric_value,
-        } = collectd_message;
+        assert_eq!(collectd_message.index(0).metric_group_key, "temperature");
```

Comment:
```suggestion
        let CollectdMessage {
            metric_group_key,
            metric_key,
            timestamp,
            metric_value,
        } = collectd_message.index(0); 
       assert_eq!(metric_group_key, "temperature");
```
Wondering if you could avoid changing all those assertion lines by doing the destructuring like this.

Reply:
updated the code.